control_plane/attachment_service/src/tenant_state.rs

```diff
@@ -688,7 +688,12 @@ impl TenantState {
             shard_count: self.tenant_shard_id.shard_count.literal() as i32,
             shard_stripe_size: self.shard.stripe_size.0 as i32,
             generation: self.generation.into().unwrap_or(0) as i32,
-            generation_pageserver: i64::MAX,
+            generation_pageserver: self
+                .intent
+                .get_attached()
+                .map(|n| n.0 as i64)
+                .unwrap_or(i64::MAX),
```

Comment:
Amazing this didn't cause any tests to fail? I guess my now-temporariliy-disabled benchmarks ([1](https://github.com/neondatabase/neon/issues/6724), [2](https://github.com/neondatabase/neon/pull/6785)) are still the only ones restarting attachment service?

Reply:
It's because this to_persistent method was added just for the consistency checks -- the actual updates to generation_pageserver happen as part of persistence::increment generation.